src/audio/module_adapter/module/volume/volume_hifi4.c

```diff
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2018 Intel Corporation. All rights reserved.
+//
+// Author: Tomasz Lauda <tomasz.lauda@linux.intel.com>
+
+/**
+ * \file
+ * \brief Volume HiFi3 processing implementation
+ * \authors Tomasz Lauda <tomasz.lauda@linux.intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/common.h>
+#include <ipc/stream.h>
+#include <stddef.h>
+#include <stdint.h>
+
+LOG_MODULE_DECLARE(volume_hifi4, CONFIG_SOF_LOG_LEVEL);
+
+#include <sof/audio/volume.h>
+
+#if VOLUME_HIFI4
```

Comment:
Generally yes, but for testing it could be practical to change build of only one component e.g. to avoid everything to slow down too much. But that can be patched too when debugging so I don't have strong opinions for this.

Reply:
We can use a local macro for this e.g.
```c
#define VOLUME_HIFI4 (HIFI4 && !VOLUME_FORCE_GENERIC)
```